browser_use/agent/message_manager/service.py

```diff
@@ -295,7 +295,10 @@ def merge_successive_human_messages(self, messages: list[BaseMessage]) -> list[B
 			if isinstance(message, HumanMessage):
 				streak += 1
 				if streak > 1:
-					merged_messages[-1].content += message.content
+					if isinstance(message.content, list):
+						merged_messages[-1].content += message.content[0]['text']
+					else:
+						merged_messages[-1].content += message.content
```

Comment:
for what is this change?

Reply:
In my testing, `deepseek-r1:7b` doesn't follow the prompt and gives you text even if you asked for a list and vice-versa. This is defensive coding based on dynamic examination of what I got from the model in pdb.

Mistral is now advocating using pydantic based validation instead of this type of `ininstance()` checks.